orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -1265,6 +1263,90 @@ void RegisterTrainingOpSchemas() {
           {"tensor(bool)"},
           "Constrain types to boolean tensors.");
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(Adam)
```

Comment:
Use description or comments to explain how this is different than AdamOptimizer. Add context around why we are adding this and some guidance to devs regarding when to use this vs AdamOptimizer.

Reply:
Wondering if we can make the Op name more intuitive, something like AdamAggregate or AggregatedAdamOptimizer... This is not the best suggestion... will thing of more names.

Reply:
I was thinking to bump up the AdamOptimizer into MSDomain=2 if possible. Because AdamOptimizer is the best one. 
I finally use "Adam" since ONNX also use this name.

For others, FusedAdam is also an option, but I don't like it to be ONNX operator name. :)

Reply:
Updated AdamWOptimizer taking suggestions from you and Baiju. :)